packages/page-staking/src/Performance/index.tsx

```diff
@@ -1,86 +1,173 @@
 // Copyright 2017-2022 @polkadot/app-staking authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
-import type { SortedTargets } from '../types';
-
-import React, { useMemo } from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 
+import { DeriveSessionProgress } from '@polkadot/api-derive/types';
+import { useTranslation } from '@polkadot/app-staking/translate';
+import { Input, Spinner } from '@polkadot/react-components';
 import { useApi, useCall } from '@polkadot/react-hooks';
-import { Vec } from '@polkadot/types';
-import { AccountId } from '@polkadot/types/interfaces';
-import { Struct, u32 } from '@polkadot/types-codec';
+import { EraIndex } from '@polkadot/types/interfaces';
+import { Option, u32 } from '@polkadot/types-codec';
 
-import ActionsBanner from './ActionsBanner';
-import CurrentList from './CurrentList';
-import Summary from './Summary';
+import Performance from './Performance';
 
 interface Props {
   className?: string;
   favorites: string[];
-  targets: SortedTargets;
   toggleFavorite: (address: string) => void;
-  session?: number,
 }
 
-interface CurrentEraValidators extends Struct {
-  reserved: Vec<AccountId>;
-  nonReserved: Vec<AccountId>;
+export interface SessionEra {
+  session: number,
+  era: number,
+  currentSessionMode: boolean,
 }
 
-type SessionValidatorBlockCountEntry = [{ args: [AccountId] }, u32];
+type SessionIndexEntry = [{ args: [EraIndex] }, Option<u32>];
 
-function Performance ({ className = '', favorites, session, targets, toggleFavorite }: Props): React.ReactElement<Props> {
+function PerformancePage ({ favorites, toggleFavorite }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
   const { api } = useApi();
-  const currentEraValidators = useCall<CurrentEraValidators>(api.query.elections.currentEraValidators);
-  const sessionValidatorBlockCountEntries = useCall<SessionValidatorBlockCountEntry[]>(api.query.elections.sessionValidatorBlockCount.entries);
+  const sessionInfo = useCall<DeriveSessionProgress>(api.derive.session.progress);
+  const historyDepth = useCall<number>(api.query.staking.historyDepth);
+  const [{ isValid, key }, setValue] = useState<{ isValid: boolean; key: string }>(() => ({
```

Comment:
i think that it can be done somehow like this: `[value, setValue] = useState(...)` and initial value set to object, but consider renaming it since setValue is not very clear

Reply:
Good point, reworked